drivers/spi/spi_mcux_lpspi.c

```diff
@@ -585,14 +608,129 @@ static int spi_mcux_init(const struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_SPI_RTIO
+static void spi_mcux_iodev_complete(const struct device *dev, int status);
+static void spi_mcux_iodev_next(const struct device *dev, bool completion);
+
+static void spi_mcux_iodev_start(const struct device *dev)
+{
+	struct spi_mcux_data *data = dev->data;
+	struct rtio_sqe *sqe = &data->txn_curr->sqe;
+	struct spi_dt_spec *dt_spec = sqe->iodev->data;
+	struct spi_buf tx_buffer;
+	struct spi_buf rx_buffer;
+	struct spi_buf_set tx;
+	struct spi_buf_set rx;
+	int ret = 0;
+
+	switch (sqe->op) {
+	case RTIO_OP_TX:
```

Comment:
This seems to be mixing transmit and receive in the same spi_buf setup which looks a bit wrong to me

Reply:
This should no longer be the case.